src/main/java/net/neoforged/neoforge/event/AddReloadListenerEvent.java

```diff
@@ -29,13 +29,14 @@
  * The event is fired on the {@link NeoForge#EVENT_BUS}
  */
 public class AddReloadListenerEvent extends Event {
-    private final List<PreparableReloadListener> listeners = new ArrayList<>();
+    private final List<PreparableReloadListener> listeners;
     private final ReloadableServerResources serverResources;
     private final RegistryAccess registryAccess;
 
-    public AddReloadListenerEvent(ReloadableServerResources serverResources, RegistryAccess registryAccess) {
+    public AddReloadListenerEvent(ReloadableServerResources serverResources, RegistryAccess registryAccess, List<PreparableReloadListener> listeners) {
         this.serverResources = serverResources;
         this.registryAccess = registryAccess;
+        this.listeners = listeners;
     }
```

Comment:
Per the compatibility check, you'll need to reinstate the old two-args constructor for binary compatibility. However, since I think no mod will need to call this constructor specifically, the old constructor should just call this new constructor with an empty list, and this new constructor should be annotated with `@ApiStatus.Internal`.